### PR TITLE
Simplify DictionaryBlock creation [spilljoin]

### PR DESCRIPTION
**Pull request by findepi**
Extracted from #8166

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void testCopyPositionsWithCompactionsAndReorder()
         assertEquals(copiedBlock.getDictionary().getPositionCount(), 2);
         assertEquals(copiedBlock.getPositionCount(), positionsToCopy.size());
 
-        assertBlock(copiedBlock.getDictionary(), new Slice[] { expectedValues[0], expectedValues[5] });
+        assertBlock(copiedBlock.getDictionary(), new Slice[] {expectedValues[0], expectedValues[5]});
```
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java]: nit: Extract formatting fixes to separate commit

File: presto-spi/src/main/java/com/facebook/presto/spi/Page.java [hunk 2]
```diff
@@ -252,6 +252,17 @@ private static int determinePositionCount(Block... blocks)
         return blocks[0].getPositionCount();
     }
 
+    public static Page mask(Page page, int[] retainedPositions)
```
Comment (losipiuk) [inline presto-spi/src/main/java/com/facebook/presto/spi/Page.java]: This is not used anywhere.

Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/Page.java]: It's not, but will be in Spill (partitioning spiller and LookupJoinOperator both need this). Should I move adding this method to that PR?

File: presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java (hunk 1 above)
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java]: Done

File: presto-spi/src/main/java/com/facebook/presto/spi/Page.java (hunk 2 above)
Comment (losipiuk) [inline presto-spi/src/main/java/com/facebook/presto/spi/Page.java]: no. Keep it IMO.

**Comment by findepi** [comment]
@dain, do you want to review this? 
(Only refactor around `DictionaryBlock`-s).

**Comment by dain** [comment]
@findepi, I skimmed it and it seems reasonable.  Please wait to merge until after this release goes out.

**Comment by findepi** [comment]
@dain thanks! Sure, I'll wait with merging, I don't have commit rights anyway :D

**Comment by findepi** [comment]
rebased on master

**Comment by losipiuk** [comment]
Merged!
